web/ui/src/Services/AppsController.js

```diff
@@ -95,32 +95,26 @@
 
             service.model.Endpoints.forEach(endpoint => {
                 if(endpoint.VHostList){
-                    endpoint.VHostList.forEach(vHost => endPoints.push(vHost));
+                    endpoint.VHostList.forEach(vHost => {
+                        vHost.URL = utils.publicEndpointURL($scope, vHost);
+                        endPoints.push(vHost);
+                    });
                 }
                 if(endpoint.PortList){
-                    endpoint.PortList.forEach(port => endPoints.push(port));
+                    endpoint.PortList.forEach(port => {
+                        port.URL = utils.publicEndpointURL($scope, port);
+                        endPoints.push(port);
+                    });
                 }
             });
 
-            endPoints.sort();
+            endPoints.sort(function(pep1, pep2) { return pep1.URL.localeCompare(pep2.URL); });
```

Comment:
I'll get that changed after lunch, no problem.
On May 5, 2016 1:08 PM, "John Matos" <notifications@github.com> wrote:

> In web/ui/src/Services/AppsController.js
> <https://github.com/control-center/serviced/pull/2698#discussion_r62228294>
> :
>
> >                  }
> >              });
> >
> > -            endPoints.sort();
> > +            endPoints.sort(function(pep1, pep2) { return pep1.URL.localeCompare(pep2.URL); });
>
> not strictly necessary, but es6 arrow functions are more succint for this
> exactly usage:
> endPoints.sort((pep1, pep2) =? pep1.URL.localeCompare(pep2.URL));
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/control-center/serviced/pull/2698/files/c157076e87224cd6e72bbcb085c4ec1e3d2997bd#r62228294>
>